### PR TITLE
src: update pack version

### LINKED ISSUE #255: Quarkus build fail results in extraneous error reporting
When the compilation of a Quarkus function fails the returned information seems to be excessively full of [builder] tags (one for every character in the return error messages.
Screenshot 2021-01-29 at 09 42 37

### PR DESCRIPTION
**Pull request by matejvasek**
resolves #255

Signed-off-by: Matej Vasek <mvasek@redhat.com>

### CODE REVIEW

**Comment by matejvasek** [comment]
@zroubalik can you please test this on mac?

**Comment by matejvasek** [comment]
Also there were some updates to lifecycle images that are important for `podman` build. We should rebuild our images with latest pack so they will pick up latest lifecycle.

**Comment by matejvasek** [comment]
this seems to fix: https://github.com/boson-project/func/issues/255

File: go.mod [hunk 1]
```diff
@@ -25,10 +25,10 @@ replace (
 
 	// This is required to pin the docker module to that version that build packs are requiring
 	// Otherwise it's overwritten by knative-dev/test-infra to a version v.1.13 that is higher
-	github.com/docker/docker => github.com/docker/docker v1.4.2-0.20200221181110-62bd5a33f707
+	// github.com/docker/docker => github.com/docker/docker v1.4.2-0.20200221181110-62bd5a33f707
 
 	// Needed for macos
-	golang.org/x/sys => golang.org/x/sys v0.0.0-20200302150141-5c8b2ff67527
+	// golang.org/x/sys => golang.org/x/sys v0.0.0-20200302150141-5c8b2ff67527
```
Comment (matejvasek) [inline go.mod:31]: Do we still need this for macos?

File: go.mod [hunk 2]
```diff
@@ -25,10 +25,10 @@ replace (
 
 	// This is required to pin the docker module to that version that build packs are requiring
 	// Otherwise it's overwritten by knative-dev/test-infra to a version v.1.13 that is higher
-	github.com/docker/docker => github.com/docker/docker v1.4.2-0.20200221181110-62bd5a33f707
+	// github.com/docker/docker => github.com/docker/docker v1.4.2-0.20200221181110-62bd5a33f707
```
Comment (matejvasek) [inline go.mod:28]: Looks this is not needed with newer version of `pack`.

File: go.mod (hunk 1 above)
Comment (matejvasek) [reply go.mod:31]: GH action for mac is passing.

**Comment by zroubalik** [comment]
> @zroubalik can you please test this on mac?

Yeah I can do that, anything specific I should test?

**Comment by matejvasek** [comment]
@zroubalik nothing specific, just test whether it can create and build Function project.

**Comment by zroubalik** [comment]
@matejvasek works for me

File: go.mod [hunk 3]
```diff
@@ -23,13 +23,6 @@ replace (
 	// packages (and old docker versions)
 	github.com/Sirupsen/logrus => github.com/sirupsen/logrus v1.7.0
```
Comment (matejvasek) [inline go.mod:24]: Maybe this also could be removed. At least removal is not causing compile time errors. @rhuss I believe you added this, was it causing compilation error with pack v0.14.0 before?

Comment (zroubalik) [reply go.mod:24]: Build working for me on Mac OS without this replace

**Comment by matejvasek** [comment]
@dsimansk I wonder can this affect integration with `kn`?

**Comment by dsimansk** [comment]
The `golang.org/x/sys` version was breaking kn cross-build on Mac OS in previously release 0.19/1.13 Serveless. I've tried to remove it. It's probably conflicting with whatever kn/Knative deps are bringing. I'll give it a try, but feel free to merge the PR as is right now. I can probably manipulate the replacements only on our side.

**Review by zroubalik — APPROVED** [review]
LGTM

File: go.mod (hunk 3 above)
Comment (zroubalik) [reply go.mod:24]: @matejvasek do you want to remove this also?

Comment (lkingland) [reply go.mod:24]: Build also working for me.  The less requires the better!

**Review by lkingland — APPROVED** [review]
LGTM too.  Nice to see a few less requires

File: go.mod (hunk 3 above)
Comment (matejvasek) [reply go.mod:24]: @zroubalik d'oh, I thought I did.

Comment (zroubalik) [reply go.mod:24]: No problem :) Fixed: https://github.com/boson-project/func/pull/258

**Pull request merged**